React/Modules/RCTNativeAnimationManager.m

```diff
@@ -0,0 +1,236 @@
+    // TODO Throw Exception
+    return;
+  }
+  
+  RCTAnimatedNode *parentNode = [self nodeForTag:parentTag];
+  RCTAnimatedNode *childNode = [self nodeForTag:childTag];
+  
+  if (!parentNode || !childNode) {
+    // Throw Exception
+    return;
+  }
+  
+  [parentNode addChild:childNode];
+}
+
+- (void)disconnectAnimatedNodes:(nonnull NSNumber *)parentTag
+                       childTag:(nonnull NSNumber *)childTag {
+  if (!parentTag || !childTag) {
+    // Throw Exception
+    return;
+  }
+  
+  RCTAnimatedNode *parentNode = [self nodeForTag:parentTag];
+  RCTAnimatedNode *childNode = [self nodeForTag:childTag];
+  
+  if (!parentNode || !childNode) {
+    // Throw Exception
+    return;
+  }
+  
+  [parentNode removeChild:childNode];
+}
+
+- (void)startAnimatingNode:(nonnull NSNumber *)animationId
+                   nodeTag:(nonnull NSNumber *)nodeTag
+                    config:(nonnull NSDictionary *)config
+               endCallback:(nullable RCTResponseSenderBlock)callBack {
+  if (![config[@"type"] isEqualToString:@"frames"]) {
+    // Exception Unsupported animation type
+    return;
+  }
+  
+  NSNumber *delay = config[@"delay"];
+  NSNumber *toValue = config[@"toValue"] ?: @1;
+  NSArray *frames = config[@"frames"];
+  
+  RCTValueAnimatedNode *valueNode = (RCTValueAnimatedNode *)[self nodeForTag:nodeTag];
+  
+  RCTAnimationDriverNode *animationDriver =
+    [[RCTAnimationDriverNode alloc] initWithId:animationId
+                                         delay:delay
+                                       toValue:toValue
+                                        frames:frames
+                                       forNode:valueNode
+                                      callBack:callBack];
+  [_activeAnimations addObject:animationDriver];
+  [_animationDrivers setObject:animationDriver forKey:animationId];
+  [animationDriver startAnimation];
+  [self startAnimation];
+}
+
+- (void)stopAnimation:(nonnull NSNumber *)animationId {
+  RCTAnimationDriverNode *driver = [_animationDrivers objectForKey:animationId];
+  if (driver) {
+    [driver removeAnimation];
+    [_animationDrivers removeObjectForKey:animationId];
+    [_activeAnimations removeObject:driver];
+    [_finishedAnimations removeObject:driver];
+  }
+}
+
+- (void)setAnimatedNodeValue:(nonnull NSNumber *)nodeTag
+                       value:(nonnull NSNumber *)value {
+  RCTAnimatedNode *node = [self nodeForTag:nodeTag];
+  if (!node ||
+      ![node isKindOfClass:[RCTValueAnimatedNode class]]) {
+    NSLog(@"Not a vlue node. Exception");
+    return;
+  }
+  RCTValueAnimatedNode *valueNode = (RCTValueAnimatedNode *)node;
+  valueNode.value = value;
+  [valueNode setNeedsUpdate];
+}
+
+- (void)connectAnimatedNodeToView:(nonnull NSNumber *)nodeTag
+                          viewTag:(nonnull NSNumber *)viewTag {
+  RCTAnimatedNode *node = [self nodeForTag:nodeTag];
+  if (viewTag && node && [node isKindOfClass:[RCTPropsAnimatedNode class]]) {
```

Comment:
No need for the && node check here - if node is nil the isKindOfClass: call will return NO anyway